packages/expo-camera/ios/Current/CameraView.swift

```diff
@@ -258,26 +257,29 @@ public class CameraView: ExpoView, EXCameraInterface, EXAppLifecycleListener,
       return
     }
 
-    sessionQueue.async {
-      let photoOutput = AVCapturePhotoOutput()
-      photoOutput.isLivePhotoCaptureEnabled = false
-      if self.session.canAddOutput(photoOutput) {
-        self.session.addOutput(photoOutput)
-        self.photoOutput = photoOutput
-      }
-
-      self.session.sessionPreset = self.mode == .video ? self.pictureSize.toCapturePreset() : .photo
-      self.addErrorNotification()
-      self.changePreviewOrientation()
+    let photoOutput = AVCapturePhotoOutput()
+    photoOutput.isLivePhotoCaptureEnabled = false
+    if session.canAddOutput(photoOutput) {
+      session.addOutput(photoOutput)
+      self.photoOutput = photoOutput
     }
 
-    // Delay starting the scanner
-    sessionQueue.asyncAfter(deadline: .now() + 0.5) {
-      self.barcodeScanner.maybeStartBarcodeScanning()
-      self.session.startRunning()
-      self.onCameraReady()
-      self.enableTorch()
+    session.sessionPreset = mode == .video ? pictureSize.toCapturePreset() : .photo
+    addErrorNotification()
+    await changePreviewOrientation()
+
+    // Delay starting the scanner. Starting it immediately will result in it not working on the first render.
+    do {
+      try await Task.sleep(seconds: 0.5)
```

Comment:
Instead of sleeping, can we just put a new task on the UI queue? That would generate a small delay, but maybe that is enough? 
```
MainActor.run {
	// staring the scanner
}
```

Reply:
After more testing it looks like we don't need this delay at all anymore